src/deferred.js

```diff
@@ -13,6 +13,23 @@ function Thrower( ex ) {
 	throw ex;
 }
 
+function adoptValue( value, resolve, reject ) {
+	var method;
+
+	// Check for promise aspect first to privilege synchronous behavior
+	if ( value && jQuery.isFunction( ( method = value.promise ) ) ) {
```

Comment:
@markelog: Commenting it out won't fail any tests because Promises/A+ only specifies `then`, and we only define a [promises-tests adapter](https://github.com/jquery/jquery/blob/055cb7534e2dcf7ee8ad145be83cb2d74b5331c7/test/promises_aplus_adapter.js#L13-L21) for `jQuery.Deferred`. I can define a `jQuery.when` adapter, put in a QUnit test of `jQuery.when( pathologicalThenable )`, or just let it be.